src/NServiceBus.Persistence.DynamoDB.Tests/Outbox/WriteRequestBatchExecutionTests.cs

```diff
@@ -0,0 +1,448 @@
+namespace NServiceBus.Persistence.DynamoDB.Tests
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Threading.Tasks;
+    using Amazon.DynamoDBv2.Model;
+    using Amazon.Runtime;
+    using Logging;
+    using NUnit.Framework;
+    using Particular.Approvals;
+
+    [TestFixture]
+    public class WriteRequestBatchExecutionTests
+    {
+        [SetUp]
+        public void SetUp()
+        {
+            configuration = new OutboxPersistenceConfiguration
+            {
+                Table = new()
+                {
+                    TableName = "faketable",
+                    PartitionKeyName = "PK",
+                    SortKeyName = "SK",
+                }
+            };
+            logger = new Logger();
+            client = new MockDynamoDBClient();
+        }
+
+        [Test]
+        public async Task ExecutesBatches()
+        {
+            var batches = new List<List<WriteRequest>>
+            {
+                new() { new WriteRequest(), new WriteRequest() },
+                new() { new WriteRequest(), new WriteRequest(), new WriteRequest() }
+            };
+
+            await client.BatchWriteItemWithRetries(batches, configuration, logger);
+
+            Assert.That(client.BatchWriteRequestsSent, Has.Count.EqualTo(2));
+            Assert.That(client.BatchWriteRequestsSent, Has.One.Matches<BatchWriteItemRequest>(x => x.RequestItems[configuration.Table.TableName].Count == 2));
+            Assert.That(client.BatchWriteRequestsSent, Has.One.Matches<BatchWriteItemRequest>(x => x.RequestItems[configuration.Table.TableName].Count == 3));
+        }
+
+        [Test]
+        public async Task LogsDetailsOnDebugLogging()
```

Comment:
do we really test log conditions or log output in other places?

Reply:
Why does it matter?